graphbolt/src/unique_and_compact.cc

```diff
@@ -0,0 +1,34 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ *
+ * @file unique_and_compact.cc
+ * @brief Unique and compact op.
+ */
+#ifndef GRAPHBOLT_UNIQUE_AND_COMPACT_H_
+#define GRAPHBOLT_UNIQUE_AND_COMPACT_H_
+
+#include "./unique_and_compact.h"
+
+#include "./concurrent_id_hash_map.h"
+
+namespace graphbolt {
+namespace sampling {
+std::tuple<torch::Tensor, torch::Tensor, torch::Tensor> Unique_and_compact(
+    const torch::Tensor& src_ids, const torch::Tensor& dst_ids,
+    const torch::Tensor unique_dst_ids) {
+  torch::Tensor compacted_src_ids;
+  torch::Tensor compacted_dst_ids;
+  torch::Tensor unique_ids;
+  auto num_dst = unique_dst_ids.size(0);
+  torch::Tensor ids = torch::cat({unique_dst_ids, src_ids});
```

Comment:
What would happen if `dst_ids` is not a subset of `unique_dst_ids`?

Reply:
Will MapIds fail?

Reply:
No, it return -1 for unfounded ids.

Reply:
Is this an expected behavior? Or we need to raise a runtime error?

Reply:
Good point, maybe we should raise an error here https://github.com/dmlc/dgl/blob/7e4c226de460e0b3ac82952ea7a075c0256f8b5a/graphbolt/src/concurrent_id_hash_map.cc#L163C21-L163C21.
If so, let's draft a separated PR for it, and also add test.

Reply:
Sure, I'll draft a separate PR and add test in this PR.